File: backend/src/main/java/com/bomnuocv1/domain/valueobject/PricingType.java
package com.bomnuocv1.domain.valueobject;

import lombok.Getter;

@Getter
public enum PricingType {
    PER_AREA("per_area", "Theo diện tích"),
    PER_HOUR("per_hour", "Theo thời gian (giờ)");

    private final String code;
    private final String description;

    PricingType(String code, String description) {
        this.code = code;
        this.description = description;
    }

    public static PricingType fromCode(String code) {
        if (code == null) {
            return PER_AREA;
        }
        for (PricingType type : values()) {
            if (type.code.equalsIgnoreCase(code.trim()) || type.name().equalsIgnoreCase(code.trim())) {
                return type;
            }
        }
        return PER_AREA;
    }
}

File: frontend/app/src/main/java/com/vn/bomnuocv1/domain/model/PricingType.kt
package com.vn.bomnuocv1.domain.model

enum class PricingType(val code: String, val description: String) {
    PER_AREA("per_area", "Theo diện tích"),
    PER_HOUR("per_hour", "Theo thời gian (giờ)");

    companion object {
        fun fromCode(code: String?): PricingType {
            if (code == null) return PER_AREA
            return entries.firstOrNull {
                it.code.equals(code.trim(), ignoreCase = true) ||
                        it.name.equals(code.trim(), ignoreCase = true)
            } ?: PER_AREA
        }
    }
}
